### PR TITLE
Remove all operational code and docs for Xray resources

### PR DESCRIPTION
**Pull request by alexhung**
### NOT TO BE MERGED UNTIL/AFTER MARCH 31ST

*Reminder:* update version to 3.0.0

Per guideline from Terraform: https://www.terraform.io/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-removal

- remove code for Xray policy and watch
- return error for Create and Read functions
- remove documentations

### CODE REVIEW

File: pkg/artifactory/resource_xray_watch.go [hunk 1]
```diff
@@ -164,199 +111,12 @@ func resourceXrayWatch() *schema.Resource {
...
-
-	watch := expandWatch(d)
-	_, err := m.(*resty.Client).R().SetBody(watch).Post("xray/api/v2/watches")
-	if err != nil {
-		return err
-	}
-
-	d.SetId(*watch.GeneralData.Name) // ID may be returned according to the API docs, but not in go-xray
-	return resourceXrayWatchRead(d, m)
+	return errors.New("Use Xray provider resource instead")
```
Comment (chb0github) [inline pkg/artifactory/resource_xray_watch.go:115]: This is nice of you, but I would literally remove everything.

Comment (alexhung) [reply pkg/artifactory/resource_xray_watch.go:115]: This is from TF guideline.

Comment (chb0github) [reply pkg/artifactory/resource_xray_watch.go:115]: Can you link we the guide?

Comment (alexhung) [reply pkg/artifactory/resource_xray_watch.go:115]: https://www.terraform.io/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-removal

**Review by chb0github — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

**Review by dasmanas — APPROVED** [review]
looks good

**Pull request merged**